src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -623,8 +624,14 @@ void GoogleCalendarSyncAdaptor::purgeDataForOldAccount(int oldId, SocialNetworkS
 void GoogleCalendarSyncAdaptor::beginSync(int accountId, const QString &accessToken)
 {
     SOCIALD_LOG_DEBUG("beginning Calendar sync for Google, account" << accountId);
-
     bool needCleanSync = !wasLastSyncSuccessful(accountId);
+    if (needCleanSync) {
```

Comment:
Not working as in not removing notebook? Not spotting any detail why it shouldn't.